packages/@aws-cdk/aws-logs/lib/log-retention-provider/index.ts

```diff
@@ -64,21 +85,10 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
       await setRetentionPolicy(logGroupName, logGroupRegion, retryOptions, parseInt(event.ResourceProperties.RetentionInDays, 10));
 
       if (event.RequestType === 'Create') {
-        // Set a retention policy of 1 day on the logs of this function. The log
-        // group for this function should already exist at this stage because we
-        // already logged the event but due to the async nature of Lambda logging
-        // there could be a race condition. So we also try to create the log group
-        // of this function first. If multiple LogRetention constructs are present
-        // in the stack, they will try to act on this function's log group at the
-        // same time. This can sometime result in an OperationAbortedException. To
-        // avoid this and because this operation is not critical we catch all errors.
-        try {
-          const region = process.env.AWS_REGION;
-          await createLogGroupSafe(`/aws/lambda/${context.functionName}`, region, retryOptions);
-          await setRetentionPolicy(`/aws/lambda/${context.functionName}`, region, retryOptions, 1);
```

Comment:
See in the old code, if CreateLogGroupSafe throws an error, setRetentionPolicy is skipped.

Reply:
What do you mean by this? 
This behaviour is the same in your updated code as well, i.e., if `createLogGroupSafe()` throws an error `setRetentionPolicy()` is skipped.

Reply:
Old code:
CreateLogGroupSafe encounters OperationAbortedException, throws exception, setRetentionPolicy is skipped, the error is eaten by catch all.
Result: retention policy for this function is not set and no error. This is a bug in the old code I'm talking about.

New code:
CreateLogGroupSafe encounters OperationAbortedException, retries, if really fails after several attempts, then the error is floated up the stack.
Result: either log group is created with limited retention, or an exception is thrown.